packages/eventual-send/src/index.js

```diff
@@ -71,33 +68,109 @@ export function makeHandledPromise() {
       while (target !== p) {
         const parent = forwardedPromiseToPromise.get(target);
         forwardedPromiseToPromise.delete(target);
-        promiseToUnsettledHandler.delete(target);
+        promiseToPendingHandler.delete(target);
         promiseToPresence.set(target, presence);
         target = parent;
       }
     } else {
-      // We propagate p and remove all other unsettled handlers
+      // We propagate p and remove all other pending handlers
       // upstream.
       // Note that everything except presences is covered here.
       while (target !== p) {
         const parent = forwardedPromiseToPromise.get(target);
         forwardedPromiseToPromise.set(target, p);
-        promiseToUnsettledHandler.delete(target);
+        promiseToPendingHandler.delete(target);
         target = parent;
       }
     }
     return target;
   }
 
-  // This special handler accepts Promises, and forwards
-  // handled Promises to their corresponding fulfilledHandler.
+  /**
+   * This special handler accepts Promises, and forwards
+   * handled Promises to their corresponding fulfilledHandler.
+   *
+   * @type {Required<EHandler<any>>}
+   */
   let forwardingHandler;
   let handle;
 
-  function HandledPromise(executor, unsettledHandler = undefined) {
-    if (new.target === undefined) {
-      throw new Error('must be invoked with "new"');
+  /**
+   * @param {string} handlerName
+   * @param {EHandler<any>} handler
+   * @param {string} operation
+   * @param {any} o
+   * @param {any[]} opArgs
+   * @param {Promise<unknown>} [returnedP]
+   * @returns {any}
+   */
+  const dispatchToHandler = (
+    handlerName,
+    handler,
+    operation,
+    o,
+    opArgs,
+    returnedP,
+  ) => {
+    let actualOp = operation;
+    const triedOperations = [operation];
+    const isSendOnly = actualOp.endsWith('SendOnly');
+    const makeResult = result => (isSendOnly ? undefined : result);
+
+    if (isSendOnly) {
+      // We don't specify the resulting promise if it is sendonly.
+      returnedP = undefined;
+    }
+
+    if (isSendOnly && typeof handler[actualOp] !== 'function') {
+      // Substitute for sendonly with the corresponding non-sendonly operation.
+      actualOp = actualOp.slice(0, -'SendOnly'.length);
```

Comment:
omg. Since you are both checking for the suffix and extracting the prefix, I'd prefer something like

```suggestion
const SendOnlyPattern = /^(.*)SendOnly$/;
...
    const match = SendOnlyPattern.exec(actualOp);
    const isSendOnly = match !== null;
    ...  
      actualOp = match[1];
```

Reply:
Yeah, I forgot that I can actually use RegExp; we're not under Jessie handcuffs here.

The irony is that I came to Javascript from Perl, so regexps are almost always more readable to me than string manipulations.